client/app/lib/nodenotificationclient.coffee

```diff
@@ -0,0 +1,136 @@
+debug = require('debug') 'app:nodenotification'
+
+kd = require 'kd'
+
+SockJS = require 'sockjs-client'
+backoff = require 'backoff'
+
+kookies = require 'kookies'
+globals = require 'globals'
+
+nick = require 'app/util/nick'
+
+
+module.exports = class NodeNotificationClient extends kd.Object
+
+  SUBSCRIBE_ENDPOINT = '/notify/subscribe' # this can be taken from config ~ GG
+
+
+  constructor: (options = {}, data) ->
+
+    super options, data
+
+
+    @_bo = backoff.exponential
+      initialDelay : 1200
+      maxDelay     : 15000
+
+    @on 'ready', =>
+      debug 'connected and verified, listening for new events...'
+      @_bo.reset()
+
+    @_bo.failAfter 15
+
+    @_bo.on 'backoff', (trial, delay) ->
+      return  if trial is 0
+      debug "failed to connect on trial #{trial}, trying again in #{delay}ms..."
+
+    @_bo.on 'fail', ->
+      debug 'failed to connect, giving up.'
+
+
+  connect: ->
+
+    @_bo.on 'ready', =>
+      @_socket?.close()
+      delete @_socket
+      @_connect()
+
+    @_bo.backoff()
+
+
+  _connect: ->
+
+    debug 'initiating the socket connection...'
+    @_socket = new SockJS SUBSCRIBE_ENDPOINT
+
+    @_socket.onmessage = @bound '_handleNewMessage'
+    @_socket.onclose   = @bound '_handleClose'
+    @_socket.onopen    = @bound '_handleOpen'
```

Comment:
We can simply log the errors first and deal with handling them as we see any.

If connection ready state is not open yet then just bind an event handler and send the message in the inline callback function. Otherwise, messages need to be stored in a queue.